### ISSUE TITLE
ImportError: cannot import name 'force_text' from 'django.utils.encoding'

### ISSUE THREAD

**Issue by zelhus-iliyas** [python-social-auth/social-app-django#386]
(env) -->:~/Desktop/oauth/dj-auth-api$ python3 manage.py runserver
Watching for file changes with StatReloader
Exception in thread django-main-thread:
Traceback (most recent call last):
  File "/usr/lib/python3.10/threading.py", line 1009, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.10/threading.py", line 946, in run
    self._target(*self._args, **self._kwargs)
  File "/home/iliyas/Desktop/oauth/dj-auth-api/env/lib/python3.10/site-packages/django/utils/autoreload.py", line 64, in wrapper
    fn(*args, **kwargs)
  File "/home/iliyas/Desktop/oauth/dj-auth-api/env/lib/python3.10/site-packages/django/core/management/commands/runserver.py", line 125, in inner_run
    autoreload.raise_last_exception()
  File "/home/iliyas/Desktop/oauth/dj-auth-api/env/lib/python3.10/site-packages/django/utils/autoreload.py", line 87, in raise_last_exception
    raise _exception[1]
  File "/home/iliyas/Desktop/oauth/dj-auth-api/env/lib/python3.10/site-packages/django/core/management/__init__.py", line 398, in execute
    autoreload.check_errors(django.setup)()
  File "/home/iliyas/Desktop/oauth/dj-auth-api/env/lib/python3.10/site-packages/django/utils/autoreload.py", line 64, in wrapper
    fn(*args, **kwargs)
  File "/home/iliyas/Desktop/oauth/dj-auth-api/env/lib/python3.10/site-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/home/iliyas/Desktop/oauth/dj-auth-api/env/lib/python3.10/site-packages/django/apps/registry.py", line 116, in populate
    app_config.import_models()
  File "/home/iliyas/Desktop/oauth/dj-auth-api/env/lib/python3.10/site-packages/django/apps/config.py", line 269, in import_models
    self.models_module = import_module(models_module_name)
  File "/usr/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/iliyas/Desktop/oauth/dj-auth-api/env/lib/python3.10/site-packages/social_django/models.py", line 14, in <module>
    from .fields import JSONField
  File "/home/iliyas/Desktop/oauth/dj-auth-api/env/lib/python3.10/site-packages/social_django/fields.py", line 7, in <module>
    from django.utils.encoding import force_text
ImportError: cannot import name 'force_text' from 'django.utils.encoding'

requirements.txt:
asgiref==3.5.2
certifi==2022.6.15
cffi==1.15.1
charset-normalizer==2.1.1
coreapi==2.3.3
coreschema==0.0.4
cryptography==37.0.4
defusedxml==0.7.1
dj-rest-auth==2.2.5
Django==4.1
django-allauth==0.51.0
django-cors-headers==3.13.0
django-templated-mail==1.1.1
djangorestframework==3.13.1
djangorestframework-simplejwt==4.8.0
djoser==2.1.0
idna==3.3
itypes==1.2.0
Jinja2==3.1.2
MarkupSafe==2.1.1
oauthlib==3.2.0
pycparser==2.21
PyJWT==2.4.0
python3-openid==3.2.0
pytz==2022.2.1
requests==2.28.1
requests-oauthlib==1.3.1
six==1.16.0
social-auth-app-django==4.0.0
social-auth-core==4.3.0
sqlparse==0.4.2
uritemplate==4.1.1
urllib3==1.26.12

**Comment by nijel** [comment]
Use  social-auth-app-django==5.0.0 instead for compatiblity with recent Django releases.

**Comment by sinisarudan** [comment]
Thanks for the quick response @nijel 
Yet, users who use [djoser](https://github.com/sunscrapers/djoser) still have to rely on the hacks like the one I've proposed.
Namely, djoser has not issued new release for some time already and it has conflicting dependency with `social-auth-app-django`
```
 Cannot install ... social-auth-app-django==5.0.0 because these package versions have conflicting dependencies.

The conflict is caused by:
    The user requested social-auth-app-django==5.0.0
    djoser 2.1.0 depends on social-auth-app-django<5.0.0 and >=4.0.0
```
It is an issues recognized in [djoser community](https://github.com/sunscrapers/djoser/issues/668)

**Comment by zelhus-iliyas** [comment]
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
djoser 2.1.0 requires social-auth-app-django<5.0.0,>=4.0.0, but you have social-auth-app-django 5.0.0 which is incompatible.



Djoser could release version ?

**Comment by zelhus-iliyas** [comment]
I have kept this work in  project aside hoping djoser solves this and working other backend work. 
> I've hacked it by:
> 
> ```
> # a hack for `social_django` that relies on the legacy `force_text` that is in django > 4.0 replaced with `force_str`:
> # https://stackoverflow.com/questions/70382084/import-error-force-text-from-django-utils-encoding
> import django
> from django.utils.encoding import force_str
> django.utils.encoding.force_text = force_str
> ```

Issue seems social_django has some conflicts when I used this hack new error occured with social_django and djoser.

below is the error:


September 21, 2022 - 04:16:16
Django version 4.1.1, using settings 'core.settings'
Starting development server at http://127.0.0.1:8000/
Quit the server with CONTROL-C.
Exception in thread django-main-thread:
Traceback (most recent call last):
  File "/home/iliyas/Desktop/backend-com/env/lib/python3.10/site-packages/django/core/servers/basehttp.py", line 47, in get_internal_wsgi_application
    return import_string(app_path)
  File "/home/iliyas/Desktop/backend-com/env/lib/python3.10/site-packages/django/utils/module_loading.py", line 30, in import_string
    return cached_import(module_path, class_name)
  File "/home/iliyas/Desktop/backend-com/env/lib/python3.10/site-packages/django/utils/module_loading.py", line 15, in cached_import
    module = import_module(module_path)
  File "/usr/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/iliyas/Desktop/backend-com/core/wsgi.py", line 16, in <module>
    application = get_wsgi_application()
  File "/home/iliyas/Desktop/backend-com/env/lib/python3.10/site-packages/django/core/wsgi.py", line 13, in get_wsgi_application
    return WSGIHandler()
  File "/home/iliyas/Desktop/backend-com/env/lib/python3.10/site-packages/django/core/handlers/wsgi.py", line 125, in __init__
    self.load_middleware()
  File "/home/iliyas/Desktop/backend-com/env/lib/python3.10/site-packages/django/core/handlers/base.py", line 40, in load_middleware
    middleware = import_string(middleware_path)
  File "/home/iliyas/Desktop/backend-com/env/lib/python3.10/site-packages/django/utils/module_loading.py", line 30, in import_string
    return cached_import(module_path, class_name)
  File "/home/iliyas/Desktop/backend-com/env/lib/python3.10/site-packages/django/utils/module_loading.py", line 15, in cached_import
    module = import_module(module_path)
  File "/usr/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/iliyas/Desktop/backend-com/env/lib/python3.10/site-packages/social_django/middleware.py", line 9, in <module>
    from django.utils.http import urlquote
ImportError: cannot import name 'urlquote' from 'django.utils.http' (/home/iliyas/Desktop/backend-com/env/lib/python3.10/site-packages/django/utils/http.py)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/lib/python3.10/threading.py", line 1009, in _bootstrap_inner
    self.run()
  File "/usr/lib/python3.10/threading.py", line 946, in run
    self._target(*self._args, **self._kwargs)
  File "/home/iliyas/Desktop/backend-com/env/lib/python3.10/site-packages/django/utils/autoreload.py", line 64, in wrapper
    fn(*args, **kwargs)
  File "/home/iliyas/Desktop/backend-com/env/lib/python3.10/site-packages/django/core/management/commands/runserver.py", line 157, in inner_run
    handler = self.get_handler(*args, **options)
  File "/home/iliyas/Desktop/backend-com/env/lib/python3.10/site-packages/django/contrib/staticfiles/management/commands/runserver.py", line 31, in get_handler
    handler = super().get_handler(*args, **options)
  File "/home/iliyas/Desktop/backend-com/env/lib/python3.10/site-packages/django/core/management/commands/runserver.py", line 78, in get_handler
    return get_internal_wsgi_application()
  File "/home/iliyas/Desktop/backend-com/env/lib/python3.10/site-packages/django/core/servers/basehttp.py", line 49, in get_internal_wsgi_application
    raise ImproperlyConfigured(
django.core.exceptions.ImproperlyConfigured: WSGI application 'core.wsgi.application' could not be loaded; Error importing module.

**Comment by nijel** [comment]
That has been fixed in 5.0.0 as well - 5bb4468f08f2055b7322ad0f3c56c399892e19bb.
